engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -895,8 +895,10 @@ private void redoPartitions(PMapping currentMapping, String note)
         addedPartitions.add(newPartition);
       } else {
         // check whether mapping was changed
+        int currentPartitionsSize = mainPC.currentPartitions.size();
         for (DefaultPartition<Operator> pi : mainPC.currentPartitions) {
-          if (pi == newPartition && pi.isModified()) {
+          if (pi == newPartition && (pi.isModified() ||
```

Comment:
@vrozov Is there a reason they need to be redeployed?
I think other case need to be redeployed
If you change partition from 
```text      
     A1->B1->C1->
 S->              U->Aggregator  
     A2->B2->C2->
```
to 
```text
    A1->B1->C1->
 S->A2->B2->C2->U->Aggregator  
    A3->B3->C3->
```